### PR TITLE
Image viewer for package demos

### PR DESCRIPTION
**Pull request by PaulHax**
- [x] Inject into outputs
- [x] Add dep to template.package.json
- [x] Update existing package.json's with new deps
- [x] Update packages/vite.config.js's
- [x] Put viewer inside the image details element. Render viewer when details is expanded
- [x] Extract image details/viewer into a `itk-image-details` component
- [x] Include utilities.js in `demo-app` package?
- [x] update itk-viewer/element to fix 2d orienation
- [x] Check all demo app for all pipelines
- [ ] Fix itk-viewer/element 2d viwere for sample datasets.

### Dev steps to rebuild demo web app

Make edits in `packages/core/typescript/itk-wasm/src/bindgen/typescript/demo`

Then in a package, ie `packages/downsample` 
```
pnpm build
```

Then in typescript dir, ie `packages/downsample/typescript`
```
pnpm start
```

### CODE REVIEW

**Comment by PaulHax** [comment]
This is worth a review. 

The `readImageDicomFileSeries` pipeline sample input does not look good. Need to update slice axis in viewer.

File: packages/compare-images/typescript/test/browser/demo-app/index.html [hunk 1]
```diff
@@ -89,7 +61,8 @@ <h3>👨‍💻 Live API Demo ✨</h3>
       <sl-details disabled id="compareImages-metrics-details" summary="metrics: Metrics for the baseline with the fewest number of pixels outside the tolerances."></sl-details>
       <sl-button variant="neutral" outline name="metrics-download" disabled>Download</sl-button>
 <br /><br />
-      <sl-details disabled id="compareImages-difference-image-details" summary="differenceImage: Absolute difference image"></sl-details>
+      <!-- <sl-details disabled id="compareImages-difference-image-details" summary="differenceImage: Absolute difference image"></sl-details> -->
```
Comment (thewtex) [inline packages/compare-images/typescript/test/browser/demo-app/index.html:64]: remove commented

File: packages/compare-images/typescript/test/browser/demo-app/index.html [hunk 2]
```diff
@@ -115,7 +88,8 @@ <h3>👨‍💻 Live API Demo ✨</h3>
       </sl-select>
       <sl-button variant="neutral" outline name="difference-image-download" disabled>Download</sl-button>
 <br /><br />
-      <sl-details disabled id="compareImages-difference-uchar-2d-image-details" summary="differenceUchar2dImage: Unsigned char, 2D difference image for rendering"></sl-details>
+      <!-- <sl-details disabled id="compareImages-difference-uchar-2d-image-details" summary="differenceUchar2dImage: Unsigned char, 2D difference image for rendering"></sl-details> -->
```
Comment (thewtex) [inline packages/compare-images/typescript/test/browser/demo-app/index.html:91]: remove commented

File: packages/core/typescript/demo-app/package.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "@itk-wasm/demo-app",
+  "version": "0.0.0",
```
Comment (thewtex) [inline packages/core/typescript/demo-app/package.json:3]: 0.1.0

File: packages/core/typescript/demo-app/package.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  "packageManager": "pnpm@8.11.0",
+  "description": "Try pipelines in a browser",
```
Comment (thewtex) [inline packages/core/typescript/demo-app/package.json:5]: Try itk-wasm pipelines in a browser

**Review by thewtex — APPROVED** [review]
Excellent!

Just a few minor issues noted inline.

File: packages/core/typescript/demo-app/package.json (hunk 4 above)
Comment (PaulHax) [reply packages/core/typescript/demo-app/package.json]: Addressed all the comments!

**Comment by thewtex** [comment]
Please rebase for CI testing.

**Comment by PaulHax** [comment]
All checks green 💚

**Pull request merged**